### PR TITLE
Issue #18435: Fix AST consistency across MethodNameCheck xdocs examples

### PR DESCRIPTION
**Pull request by ekansh0unofficial**
Issue: #18435 

This PR improves AST consistency across MethodNameCheck xdocs examples by aligning examples with equivalent semantic intent. 

- Example 1 , 2 , 5 ,6 and 7 are normalized to share a consistent AST structure while preserving their original configurational behaviour. 
- Example 3 , 4 are still remain suppressed, as they demonstrate constructor-specific behaviour and should not be normalized with method-only examples.

### CODE REVIEW

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
>Example of Pull Request with diff image to prove that content of examples is same https://github.com/checkstyle/checkstyle/pull/18505

please share images of diff to speedup validation.

**Comment by romani** [comment]
>Example 3 , 4 are still remain suppressed, as they demonstrate constructor-specific behaviour and should not be normalized with method-only examples.

please move them to end of suppression list and make comment above `// contains ExampleX constructors` so it will be as permanente excuse to not follow same AST in Examples.

**Comment by ekansh0unofficial** [comment]
Example 1 & 2 : https://www.diffchecker.com/TUeNdpAr/
Example 1 & 5 : https://www.diffchecker.com/IVx0rPgS/
Example 1 & 6 : https://www.diffchecker.com/EANiGiRD/
Example 1 & 7 : https://www.diffchecker.com/pvzpSCJb/

Note: 
Identifier text (e.g. methodX vs MethodX) differs across examples.
The AST consistency test validates only token structure, so this does not affect consistency.
Please let me know if identifier text differences were not intended to be ignored by the AST consistency test; I’m happy to adjust accordingly.

**Comment by romani** [comment]
>Example 1 & 2 :


Good 

All other should be same by java code, same method names, ... . Only comments should be different.

**Comment by ekansh0unofficial** [comment]
Example 1v2 : https://www.diffchecker.com/GII3Fr7F/
Example 1v5 : https://www.diffchecker.com/JAMBgG0G/
Example 1v6 : https://www.diffchecker.com/aOjozfhY/
Example 1v7 : https://www.diffchecker.com/crJvBQbJ/

All the identifier names have been made consistent.

**Comment by ekansh0unofficial** [comment]
@romani  could you please check if any changes are required. Thank you.

**Comment by romani** [comment]
GitHub, generate website

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java [hunk 1]
```diff
@@ -319,8 +313,11 @@ public class XdocsExamplesAstConsistencyTest {
             "filters/suppresswithplaintextcommentfilter/Example5",
             "filters/suppresswithplaintextcommentfilter/Example9",
             // No properties in module, multiple very different examples to ease reading
-            "checks/annotation/missingoverrideonrecordaccessor/Example2"
-    );
+            "checks/annotation/missingoverrideonrecordaccessor/Example2",
+            // contains ExampleX constructors
+            "checks/naming/methodname/Example3",
+            "checks/naming/methodname/Example4"
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java:319]: @smita1078 , I thinks we can improve ast diff test to exclude constructors too, right ?

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java:319]: https://checkstyle.org/checks/naming/methodname.html#Example4-code

but may be this is very crazy edge case that does not happen often.
Lets think on this, if there will be more cases like this.

Comment (smita1078) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java:319]: > @smita1078 , I thinks we can improve ast diff test to exclude constructors too, right ?

Yes , we can

Comment (ekansh0unofficial) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java:319]: Thanks for the discussion, this is an interesting point.

I wanted to share one concern regarding excluding constructors from the AST diff. The current AST consistency test is valuable (at least from my understanding) because it is a strict, structure-preserving comparison. If we start filtering out specific node kinds (like constructors), we slightly weaken the guarantees of the test and may risk hiding real structural inconsistencies in future examples.

Of course, this is just my perspective — happy to follow whatever direction you think is best.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExamplesAstConsistencyTest.java:319]: We already doing this with class name, as it is not avoidable. Ctor name is named by class.

Let's not change AST validation for now, please put  such cases to bottom of list with comment why it becomes persistent suppression.
